core/base/inc/Rtypes.h

```diff
@@ -253,7 +233,7 @@ class ClassDefGenerateInitInstanceLocalInjector:
       static const char* Name() {
          static std::string gName;
          if (gName.empty())
-            SetName(TTypeNameExtraction<T>::Get(), gName);
+            SetName(GetDemangledName(typeid(T).name()), gName);
```

Comment:
Rather than creating a temporary std::string, can we pass a const char* or better yet the type_info (& or *) ?

Reply:
I tried to maintain the same interfaces more or less. I will change to pass the `type_info` directly and then merge this into master.